cypress/e2e/initial.spec.js

```diff
@@ -0,0 +1,157 @@
+		cy.intercept({ method: 'POST', url: '**/session/*/push' }).as('push')
+		cy.intercept({ method: 'POST', url: '**/session/*/sync' }).as('sync')
+
+		let readToken = null
+		let writeToken = null
+		cy.shareFile('/test.md')
+			.then((token) => {
+				readToken = token
+				cy.logout()
+				cy.visit(`/s/${readToken}`)
+			})
+			.then(() => {
+				// Open read only for the first time
+				cy.getEditor().should('be.visible')
+				cy.getContent()
+					.should('contain', 'Hello world')
+					.find('h2').should('contain', 'Hello world')
+
+				// Open read only for the second time
+				cy.reload()
+				cy.getEditor().should('be.visible')
+				cy.getContent()
+					.should('contain', 'Hello world')
+					.find('h2').should('contain', 'Hello world')
+
+				cy.login(user)
+				cy.shareFile('/test.md', { edit: true })
+					.then((token) => {
+						writeToken = token
+						// Open write link and edit something
+						cy.visit(`/s/${writeToken}`)
+						cy.getEditor().should('be.visible')
+						cy.getContent()
+							.should('contain', 'Hello world')
+							.find('h2').should('contain', 'Hello world')
+						cy.getContent()
+							.type('Something new {end}')
+						cy.wait('@push')
+						cy.wait('@sync')
+
+						// Reopen read only link and check if changes are there
+						cy.visit(`/s/${readToken}`)
+						cy.getEditor().should('be.visible')
+						cy.getContent()
+							.find('h2').should('contain', 'Something new Hello world')
+
+					})
+			})
+	})
+
+	it('Load after state has been saved', function() {
+		cy.uploadFile('test.md', 'text/markdown')
+		cy.intercept({ method: 'POST', url: '**/session/*/push' }).as('push')
+		cy.intercept({ method: 'POST', url: '**/session/*/sync' }).as('sync')
+		cy.intercept({ method: 'POST', url: '**/session/*/save' }).as('save')
+
+		let readToken = null
+		let writeToken = null
+		cy.shareFile('/test.md', { edit: true })
+			.then((token) => {
+				writeToken = token
+				cy.logout()
+				cy.visit(`/s/${writeToken}`)
+			})
+			.then(() => {
+				// Open read only for the first time
+				cy.getEditor().should('be.visible')
+				cy.getContent()
+					.should('contain', 'Hello world')
+					.find('h2').should('contain', 'Hello world')
+				cy.getContent()
+					.type('Something new {end}')
+				cy.get('.save-status button').click()
+				cy.wait('@save')
+
+				// Open read only for the second time
+				cy.reload()
+				cy.getEditor().should('be.visible')
+				cy.getContent()
+					.should('contain', 'Hello world')
+					.find('h2').should('contain', 'Something new Hello world')
+
+                cy.login(user)
+				cy.shareFile('/test.md')
+					.then((token) => {
+						readToken = token
+						cy.logout()
+						cy.visit(`/s/${readToken}`)
+					})
+					.then(() => {
+						// Open read only for the first time
+						cy.getEditor().should('be.visible')
+						cy.getContent()
+							.should('contain', 'Hello world')
+							.find('h2').should('contain', 'Hello world')
```

Comment:
Shouldn't this check for `Something new Hello world`?

Reply:
TODO: Double check that previous sessions are closed